tapir/coop/models.py

```diff
@@ -719,3 +718,63 @@ def populate(
             old_frozen=old_frozen,
             new_frozen=new_frozen,
         )
+
+class ResignedMembership(models.Model):
+    share_owner = models.ForeignKey(
+        ShareOwner, on_delete=models.deletion.CASCADE, verbose_name=_("Shareowner"), related_name="share_owner",
+    )
+    cancellation_date = models.DateField(
+        default=timezone.now,
+        blank=True,
+        )
+    pay_out_day = models.DateField(null=True)
+    cancellation_reason = models.CharField(max_length = 1000)
+    coop_buys_shares_back = models.BooleanField()
+    willing_to_gift_shares_to_coop = models.BooleanField()
```

Comment:
I would have a field with the 3 text options, then next to it the `transfering_shares_to` that is null if the share is not being transfered.
I was thinking it makes it clear that it can only be one of those three cases, since you can't have  both `coop_buys_shares_back` and `willing_to_gift_shares_to_coop`

Reply:
@Theophile-Madet I was stuck today, 'cause I think I made a logical mistake by declaring share_owner and transfering_shares_to as ForeignKey. I think they should be a One-To-One-Relation, 'cause there is only one ShareOwner for each MembershipResignation and also only one ShareOwner to transfer the shares. And I don't get it done to use the `related_name="owner_to_transfer"` in the template resignedmembership_confirmation_body.html. I tried:
`{% get_display_name_full transfering_shares_to %}` and also `{% get_display_name_full owner_to_tranfser %}`. Nut it is not working at all. I will try more tomorrow. I think I have to sleep one night over this ;)